internal/app/client/conformance_client.go

```diff
@@ -218,6 +221,118 @@ func (i *invoker) clientStream(
 	}, nil
 }
 
+func (i *invoker) bidiStream(
+	ctx context.Context,
+	req *v1alpha1.ClientCompatRequest,
+) (*v1alpha1.ClientResponseResult, error) {
+	stream := i.client.BidiStream(ctx)
+
+	// Add the specified request headers to the request
+	app.AddHeaders(req.RequestHeaders, stream.RequestHeader())
+
+	firstSend := true
+	fullDuplex := false
+
+	var protoErr *v1alpha1.Error
+	var headers []*v1alpha1.Header
+	var trailers []*v1alpha1.Header
+	payloads := make([]*v1alpha1.ConformancePayload, 0, 1)
+
+	for _, msg := range req.RequestMessages {
+		if err := ctx.Err(); err != nil {
+			// If an error was returned, convert it to a proto Error
+			protoErr = app.ConvertErrorToProtoError(err)
+			break
+		}
+		bsr := &v1alpha1.BidiStreamRequest{}
+		if err := msg.UnmarshalTo(bsr); err != nil {
+			return nil, err
+		}
+		if firstSend {
+			fullDuplex = bsr.FullDuplex
+		}
+		if err := stream.Send(bsr); err != nil && errors.Is(err, io.EOF) {
+			// TODO - Is this correct? Docs say the returned error will wrap EOF
+			// and clients should call Receive to get the error so assuming this error
+			// is the unwrapped error from the send call?
+			// Call receive to get the error
```

Comment:
Yep, this looks right. The only thing I'd suggest, just to be double-extra-safe, is to use the send error just in case the call to `Receive` returns nil (it shouldn't... but just in case).

Reply:
BTW, this is an interesting case that isn't adequately covered in `ClientResponseResult`: we should probably add a field to that message to indicate the number of messages that were in the test case request definition but that _couldn't be sent_, due to an error occurring before finishing the upload. (Technically, we'd want that number for client stream operations, too.)

Reply:
Added in the extra check and also added a TODO to the proto for `ClientResponseResult` that we may want that field for unsent messages.